### ISSUE TITLE
Doesn't work with xembedsni proxy

### ISSUE THREAD

**Issue by ilya-fedin** [AyatanaIndicators/ayatana-indicator-application#1]
I'm trying to use this indicator with https://github.com/KDE/plasma-workspace/tree/master/xembed-sni-proxy
But it doesn't work because of this check https://github.com/AyatanaIndicators/ayatana-indicator-application/blob/e86b0052ef26d608fde8edcb52fcc133b5bc355f/src/application-service-appstore.c#L491
xembedsniproxy items don't have a Menu property.
For example, it works with https://gitlab.com/vala-panel-project/xfce4-sntray-plugin

**Comment by sunweaver** [comment]
Hi!

> I'm trying to use this indicator with https://github.com/KDE/plasma-workspace/tree/master/xembed-sni-proxy
> But it doesn't work because of this check https://github.com/AyatanaIndicators/ayatana-indicator-application/blob/master/src/application-service-appstore.c#L489
> xembedsniproxy items don't have Menu and IconName properties.
> For example, it works with https://gitlab.com/vala-panel-project/xfce4-sntray-plugin

I just asked for plasma-workspace in Debian to split out the xembed-sni-proxy binary.

For getting this issue here fixed, do you know if applications injected via xembed-sni-proxy have any other properties (or whatever), so that they can be recognized as proxied xembed applications?

I will look into this shortly...

**Comment by ilya-fedin** [comment]
> For getting this issue here fixed, do you know if applications injected via xembed-sni-proxy have any other properties (or whatever), so that they can be recognized as proxied xembed applications?

No, but they don't have properties that other applications have
![image](https://user-images.githubusercontent.com/17829319/87204815-565e0600-c2f5-11ea-9239-bf24c5bbd6ae.png)

When I created this issue, I didn't know that the fact that indicator-application doesn't support IconPixmap is a feature. So, if the issue shouldn't be "fixed", I can just close it :)

**Comment by ilya-fedin** [comment]
Btw, I tested Xubuntu 22.04 some time ago and it's a funny thing that it actually runs both: ayatana-indicator-application-service and Xfce's default applet, it's actually just a coincidence that the default applet runs first and takes over, otherwise you would see ayatana-indicator-application-service just like in Ubuntu MATE.

**Comment by slytomcat** [comment]
https://github.com/AyatanaIndicators/libayatana-appindicator/issues/47 - another issue related to this one.

**Comment by sunweaver** [comment]
Applications still using xembed should really be rewritten, ideally against the new libayatana-appindicator-glib implementation.

ayatana-indicator-application will see support for the newer appindicators and the older appindicators in 1-2 months. Currently, I lack the time for merging / reviewing https://github.com/AyatanaIndicators/ayatana-indicator-application/pull/25 and https://github.com/AyatanaIndicators/ayatana-indicator-application/pull/24.

Please also note, that my co-maintainer of Ayatana Indicators has recently decided to disappear from the internet, so @tari01 is not available anymore, unfortunately.

**Comment by sunweaver** [comment]
I plan to continue his work here, but not at his speed...

**Comment by sunweaver** [comment]
That said, feel free to send a PR to fix this issue. Don't expect this being dealt with with high prio.

**Comment by ilya-fedin** [comment]
@sunweaver the problem is not with xembed apps themselves but with xembedsniproxy exactly and generally ayatana-indicator-application's SNI support being incomplete.

Using xembedsniproxy is the easiest way to reproduce the issue since it always uses IconPixmap but you can also look at Qt sources and see that it has a workaround especially for indicator-application by finding it running in process list (which fails sandboxed and leads to broken icons). IconPixmap is also used by last Chromium/Electron releases but not sure whether it has any workaround for indicator-application.

**Comment by ghost** [comment]
@sunweaver I see you've merged https://github.com/AyatanaIndicators/ayatana-indicator-application/pull/26 - please create a new release so distributions can benefit from it. Thanks. :)
